integreat_cms/xliff/utils.py

```diff
@@ -319,6 +319,10 @@ def get_xliff_import_diff(request, xliff_dir):
                 page=page_translation.page,
                 language=page_translation.language,
             )
+            if existing_translation in [diff_item["existing"] for diff_item in diff]:
+                duplicate_message = _("Duplicate page translation in XLIFF, skipping")
+                messages.warning(request, f"{duplicate_message} {xliff_file}")
+                continue
```

Comment:
Maybe we should place this check in the `xliffs_to_pages` function? If we only adapt the diff, I think the actual translations which are imported will still contain the duplicate...

Reply:
Yes, indeed the better place. Also this iterator would not be necessary there.

Reply:
... on the other hand we would have to introduce higher level functionality to the `xliff_to_pages` function. Currently, it does not know anything about page translations, existing versions, etc. This could work by comparing the raw values of the page id and translation language code. Maybe it is actually better to check in both locations where there already is some more context?

Reply:
Ah ok true, yeah or we check it only in `get_xliff_import_diff()` and `xliff_import_confirm()` and accept the fact that `xliff_to_pages()` might return duplicates?

Reply:
I think I found a somewhat satisfying solution: We warn the user in the diff view about the duplicate and then in the actual import catch the error and print it to the log file. This has the generic advantage that just because saving one page translations fails the full XLIFF import is not aborted.

![image](https://user-images.githubusercontent.com/15197000/211645155-85a6dece-cb39-4da4-9150-1d942cd5099f.png)

![image](https://user-images.githubusercontent.com/15197000/211645112-e6761156-166e-4300-8d87-c81ebfa981b8.png)

```
Jan 10 19:35:09 ERROR integreat_cms.xliff.utils - Error when importing new new version for <PageTranslation (id: 85, page_id: 23, language: en, slug: interpret-and-translate)> from 'augsburg_de_en_23_1_dolmetschen-und-übersetzen.xliff' by <SimpleLazyObject: <User (id: 1, username: root, team: CMS team, superuser)>>: (IntegrityError('duplicate key value violates unique constraint "pagetranslation_unique_version"\nDETAIL:  Key (page_id, language_id, version)=(23, 2, 2) already exists.\n'))
```